app/models/questionnaire_response.rb

```diff
@@ -0,0 +1,50 @@
+class QuestionnaireResponse < ApplicationRecord
+  include Mergeable
+
+  belongs_to :questionnaire
+  belongs_to :facility
+  belongs_to :user
+
+  scope :for_sync, -> {
+    with_discarded
+      .joins(:questionnaire)
+      .select("questionnaires.questionnaire_type, questionnaire_responses.*")
+  }
+
+  validates :questionnaire_id, presence: true
+  validates :facility_id, presence: true
+  validates :device_created_at, presence: true
+  validates :device_updated_at, presence: true
+  validate :facility_change
+
+  def facility_change
+    questionnaire_response = QuestionnaireResponse.with_discarded.find_by(id: id)
+    if questionnaire_response && facility_id != questionnaire_response.facility_id
+      errors.add(:facility_id, "cannot be changed for a questionnaire response")
+    end
+  end
+
+  class << self
+    def merge(attributes)
+      new_record = new(attributes)
+
+      QuestionnaireResponse.transaction do
+        existing_record = lock.with_discarded.find_by(id: attributes["id"])
```

Comment:
Since `content` is "shared mutable state" being read and modified across requests, we take a lock on the row to prevent race conditions. 
For example:
- Request 1 sends `content: {htn_male: 1}`
- At the same time Request 2 sends `content: {htn_female: 2}`
- In the absence of a lock, depending on the order of the requests, `content` will finally be either `{htn_male: 1}` or `{htn_female: 2}`

We want from all requests to be merged, and the end result should be
```
{htn_male: 1, htn_female: 2}
```